### PR TITLE
OADP-3054: BSL default configuration validation

### PR DESCRIPTION
**Pull request by sseago**

### CODE REVIEW

File: controllers/bsl.go [hunk 1]
```diff
@@ -65,11 +71,22 @@ func (r *DPAReconciler) ValidateBackupStorageLocations(dpa oadpv1alpha1.DataProt
...
+			}
 		}
 		if bslSpec.CloudStorage != nil && bslSpec.Velero != nil {
 			return false, fmt.Errorf("must choose one of bucket or velero")
 		}
 	}
+	if numDefaultLocations > 1 {
+		return false, fmt.Errorf("Only one Storage Location be set as default")
+	}
+	if numDefaultLocations == 0 && !dpa.Spec.Configuration.Velero.NoDefaultBackupLocation {
```
Comment (kaovilai) [inline controllers/bsl.go:87]:
```suggestion
	if numDefaultLocations == 0 && !(dpa.Spec.Configuration.Velero.NoDefaultBackupLocation || dpa.Spec.Configuration.Velero.HasFeatureFlag("no-secret")) {
```

just aligning with recent PR condition.

https://github.com/openshift/oadp-operator/pull/1216/files

Comment (sseago) [reply controllers/bsl.go:87]: @kaovilai That PR has been moved to 1.3.2 and is on hold, so once this PR merges, that PR can be updated with the change you suggested here.

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]

**Pull request merged**